lib/cli/run-helpers.js

```diff
@@ -137,7 +139,7 @@ exports.runMocha = (mocha, options) => {
   };
 
   if (watch) {
-    watchRun(mocha, {ui}, fileCollectParams);
+    watchRun(mocha, {ui, watchFiles, watchIgnore}, fileCollectParams);
```

Comment:
`ui` is not required here.

Reply:
https://github.com/mochajs/mocha/pull/3980/commits/1fd64f8821a17ccb5296d6d31f6125a100e7b0bd